datastore/user_profile.go

```diff
@@ -0,0 +1,138 @@
+package datastore
+
+import (
+	"database/sql"
+	"fmt"
+
+	fpr "github.com/fluidkeys/fluidkeys/fingerprint"
+	"github.com/fluidkeys/fluidkeys/pgpkey"
+	"github.com/gofrs/uuid"
+)
+
+// UserProfile represents data in the user_profiles table
+type UserProfile struct {
+	UUID                       uuid.UUID
+	OptoutEmailsExpiryWarnings bool
+	KeyID                      int
+
+	Key *pgpkey.PgpKey
+}
+
+func (up *UserProfile) loadKey(txn *sql.Tx) error {
+	var err error
+
+	if up.KeyID == 0 {
+		panic("keyID has not been set")
+	}
+	up.Key, err = loadKey(txn, up.KeyID)
+	return err
+}
+
+func loadUserProfile(txn *sql.Tx, keyID int) (*UserProfile, error) {
+	profile := UserProfile{}
+
+	var count int
+	err := transactionOrDatabase(txn).QueryRow(
+		"SELECT COUNT(*) FROM keys WHERE id=$1", keyID,
+	).Scan(&count)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if count == 0 {
+		return nil, fmt.Errorf("no such key with id %d", keyID)
+	}
+
+	query := `SELECT user_profiles.uuid,
+                     user_profiles.optout_emails_expiry_warnings,
+					 user_profiles.key_id
+			  FROM user_profiles 
+			  WHERE user_profiles.key_id=$1`
+
+	err = transactionOrDatabase(txn).QueryRow(query, keyID).Scan(
+		&profile.UUID,
+		&profile.OptoutEmailsExpiryWarnings,
+		&profile.KeyID,
+	)
+	if err == sql.ErrNoRows {
+		// no user profile found: create one
```

Comment:
That's cool - I was wondering how you were going to approach creating the user profiles!